src/coordinate_search.jl

```diff
@@ -0,0 +1,65 @@
+export coordinate_search
+
+"""
+	The coordinate search is a derivative free optimization method.
+	It minimizes a certain function f by walking on a grid and taking a step where the function values are smaller.
+	When it can no longer find any smaller values, it reduces the grid by a factor of 2, repeating the process.
```

Comment:
When it can no longer find small values in a neighbourhood, it reduces the grid size, repeating the process.
We leave out the factor 2 since we can use as input parameter later.